src/Libraries/Nop.Services/Catalog/ProductService.cs

```diff
@@ -1766,6 +1767,17 @@ public virtual async Task<string[]> GetNotExistingProductsAsync(string[] product
             return queryFilter.Except(filter).ToArray();
         }
 
+        public async Task<Product> GetByTag(string tag)
+        {
+            var tagEntity = await _productTagRepository.Table.FirstOrDefaultAsync(x => x.Name == tag);
```

Comment:
this by design can have multiple products, so return all of them, and consume them seprately